zulipterminal/model.py

```diff
@@ -300,7 +300,7 @@ def get_messages(self, *,
             if first_anchor and response['anchor'] != 10000000000000000:
                 self.index['pointer'][str(self.narrow)] = response['anchor']
             query_range = num_after + num_before + 1
-            if len(response['messages']) < (query_range):
+            if len(response['messages']) <= query_range:
```

Comment:
This seems like a reasonable fix.

Reply:
Now that it's all coming back to me, with this fix, we will be appending messages that are not immediately next to the last message.